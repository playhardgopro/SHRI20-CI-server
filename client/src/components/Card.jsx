import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import Moment from 'react-moment'
import 'moment/locale/ru'
import { IconBox, Icon } from '.'
import './scss/Card.scss'

const cn = withNaming({ n: '', e: '__', m: '_' })

const Card = ({ children, className, options }) => {
  const { buildNumber, commitMessage, commitHash, branchName, authorName, status, start, duration } = options
  // console.log(status, 'status')
  // const startDate = parseJSON(start) // local TZ
  // const durationFormatted = format(duration, 'h:m')
  // console.log(startDate)
  const cnCard = cn('card')(className)
  const cnText = cn('text')
  return (
    <div className="card">
      <div className="card__token">
        <IconBox textStyle={{ view: status.toLowerCase() }}>
          <Icon name={status.toLowerCase()} />
        </IconBox>
      </div>
      <div className="card__content">
        <div className="card__history list">
          <div className="card__status list__item">
            <div className={`card__number ${cnText({ size: '18-20', view: status.toLowerCase() })}`}>
              {`#${buildNumber}`}
            </div>
            <div className="card__message text text_size_15-20 text_view_truncate">{commitMessage}</div>
          </div>
          <div className="list__item meta meta_m-distribute_vertical">
            <div className="meta__item">
              <div className="icon-box">
                <div className="icon icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost">
                  <div className="icon">
                    <svg width="16" height="8" viewBox="0 0 16 8" xmlns="http://www.w3.org/2000/svg">
                      <path d="M3.2 4C3.2 4.27 3.2225 4.5375 3.265 4.8H0.3C0.135 4.8 0 4.665 0 4.5V3.5C0 3.335 0.135 3.2 0.3 3.2H3.265C3.2225 3.4625 3.2 3.73 3.2 4ZM15.7 3.2H12.735C12.78 3.4625 12.8 3.73 12.8 4C12.8 4.27 12.7775 4.5375 12.735 4.8H15.7C15.865 4.8 16 4.665 16 4.5V3.5C16 3.335 15.865 3.2 15.7 3.2ZM8 1.2C7.2525 1.2 6.55 1.4925 6.02 2.02C5.49 2.55 5.2 3.2525 5.2 4C5.2 4.7475 5.49 5.45 6.02 5.98C6.55 6.5075 7.2525 6.8 8 6.8C8.7475 6.8 9.45 6.5075 9.98 5.98C10.51 5.45 10.8 4.7475 10.8 4C10.8 3.2525 10.51 2.55 9.98 2.02C9.45 1.4925 8.7475 1.2 8 1.2ZM8 0C10.21 0 12 1.79 12 4C12 6.21 10.21 8 8 8C5.79 8 4 6.21 4 4C4 1.79 5.79 0 8 0Z" />
                    </svg>
                  </div>
                </div>
                <div className=" text text_size_13-16">
                  {branchName}
                  <a className="text text_view_ghost">{commitHash.slice(0, 7)}</a>
                </div>
              </div>
            </div>
            <div className="meta__item meta__item_indent-b_8">
              <div className="icon-box">
                <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost">
                  <div className="icon icon_view_ghost">
                    <svg width="13" height="14" viewBox="0 0 13 14" xmlns="http://www.w3.org/2000/svg">
                      <path d="M6.125 7C8.0582 7 9.625 5.4332 9.625 3.5C9.625 1.5668 8.0582 0 6.125 0C4.1918 0 2.625 1.5668 2.625 3.5C2.625 5.4332 4.1918 7 6.125 7ZM8.575 7.875H8.11836C7.51133 8.15391 6.83594 8.3125 6.125 8.3125C5.41406 8.3125 4.74141 8.15391 4.13164 7.875H3.675C1.64609 7.875 0 9.52109 0 11.55V12.6875C0 13.4121 0.587891 14 1.3125 14H10.9375C11.6621 14 12.25 13.4121 12.25 12.6875V11.55C12.25 9.52109 10.6039 7.875 8.575 7.875Z" />
                    </svg>
                  </div>
                </div>
                <div className="text text_size_13-16">{authorName}</div>
              </div>
            </div>
          </div>
        </div>
        <div className="card__meta card__meta_m-hr_top meta meta_distribute_vertical meta_m-distribute_horizontal text text_size_13-16 text_view_ghost">
          <div className="meta__item meta__item_indent-b_8">
            <div className="icon-box">
              <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
                <div className="icon">
                  <svg width="14" height="16" viewBox="0 0 14 16" xmlns="http://www.w3.org/2000/svg">
                    <path d="M4.60806 9H3.36264C3.15714 9 2.98901 8.83125 2.98901 8.625V7.375C2.98901 7.16875 3.15714 7 3.36264 7H4.60806C4.81355 7 4.98169 7.16875 4.98169 7.375V8.625C4.98169 8.83125 4.81355 9 4.60806 9ZM7.9707 8.625V7.375C7.9707 7.16875 7.80256 7 7.59707 7H6.35165C6.14615 7 5.97802 7.16875 5.97802 7.375V8.625C5.97802 8.83125 6.14615 9 6.35165 9H7.59707C7.80256 9 7.9707 8.83125 7.9707 8.625ZM10.9597 8.625V7.375C10.9597 7.16875 10.7916 7 10.5861 7H9.34066C9.13517 7 8.96703 7.16875 8.96703 7.375V8.625C8.96703 8.83125 9.13517 9 9.34066 9H10.5861C10.7916 9 10.9597 8.83125 10.9597 8.625ZM7.9707 11.625V10.375C7.9707 10.1687 7.80256 10 7.59707 10H6.35165C6.14615 10 5.97802 10.1687 5.97802 10.375V11.625C5.97802 11.8313 6.14615 12 6.35165 12H7.59707C7.80256 12 7.9707 11.8313 7.9707 11.625ZM4.98169 11.625V10.375C4.98169 10.1687 4.81355 10 4.60806 10H3.36264C3.15714 10 2.98901 10.1687 2.98901 10.375V11.625C2.98901 11.8313 3.15714 12 3.36264 12H4.60806C4.81355 12 4.98169 11.8313 4.98169 11.625ZM10.9597 11.625V10.375C10.9597 10.1687 10.7916 10 10.5861 10H9.34066C9.13517 10 8.96703 10.1687 8.96703 10.375V11.625C8.96703 11.8313 9.13517 12 9.34066 12H10.5861C10.7916 12 10.9597 11.8313 10.9597 11.625ZM13.9487 3.5V14.5C13.9487 15.3281 13.2793 16 12.4542 16H1.49451C0.669414 16 0 15.3281 0 14.5V3.5C0 2.67188 0.669414 2 1.49451 2H2.98901V0.375C2.98901 0.16875 3.15714 0 3.36264 0H4.60806C4.81355 0 4.98169 0.16875 4.98169 0.375V2H8.96703V0.375C8.96703 0.16875 9.13517 0 9.34066 0H10.5861C10.7916 0 10.9597 0.16875 10.9597 0.375V2H12.4542C13.2793 2 13.9487 2.67188 13.9487 3.5ZM12.4542 14.3125V5H1.49451V14.3125C1.49451 14.4156 1.57857 14.5 1.68132 14.5H12.2674C12.3701 14.5 12.4542 14.4156 12.4542 14.3125Z" />
                  </svg>
                </div>
              </div>
              <div className="">
                <Moment format="D MMM, HH:mm" locale="ru" local>
                  {start}
                </Moment>
              </div>
            </div>
          </div>
          <div className="meta__item meta__item_indent-b_8">
            {duration && (
              <div className="icon-box">
                <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
                  <div className="icon">
                    <svg width="14" height="16" viewBox="0 0 14 16" xmlns="http://www.w3.org/2000/svg">
                      <path d="M12.2788 5.75L12.9833 5.04375C13.1299 4.89687 13.1299 4.65937 12.9833 4.5125L12.4534 3.98125C12.3069 3.83438 12.07 3.83438 11.9235 3.98125L11.2782 4.62813C10.3087 3.76875 9.08365 3.19375 7.73076 3.04063V1.5H8.60359C8.80933 1.5 8.97766 1.33125 8.97766 1.125V0.375C8.97766 0.16875 8.80933 0 8.60359 0H5.36164C5.1559 0 4.98757 0.16875 4.98757 0.375V1.125C4.98757 1.33125 5.1559 1.5 5.36164 1.5H6.23447V3.04375C3.00499 3.41563 0.498718 6.1625 0.498718 9.5C0.498718 13.0906 3.40089 16 6.98262 16C10.5643 16 13.4665 13.0906 13.4665 9.5C13.4665 8.10312 13.027 6.80937 12.2788 5.75ZM6.98262 14.5C4.22696 14.5 1.995 12.2625 1.995 9.5C1.995 6.7375 4.22696 4.5 6.98262 4.5C9.73827 4.5 11.9702 6.7375 11.9702 9.5C11.9702 12.2625 9.73827 14.5 6.98262 14.5ZM7.35669 11H6.60855C6.40281 11 6.23447 10.8313 6.23447 10.625V6.375C6.23447 6.16875 6.40281 6 6.60855 6H7.35669C7.56243 6 7.73076 6.16875 7.73076 6.375V10.625C7.73076 10.8313 7.56243 11 7.35669 11Z" />
                    </svg>
                  </div>
                </div>
                <div className="text text_size_13-16 text_view_ghost">
                  <Moment duration={start} date={start + duration} locale="ru" local />
                </div>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  )
}

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Card.defaultProps = {
  children: '',
  className: {},
}

export default Card
