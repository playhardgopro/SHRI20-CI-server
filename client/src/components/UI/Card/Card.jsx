import React from 'react'
import PropTypes from 'prop-types'
import { withNaming } from '@bem-react/classname'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import Moment from 'react-moment'
import { IconBox, Icon } from '../../'
import './Card.scss'

const prettyMilliseconds = require('pretty-ms')

const cn = withNaming({ n: '', e: '__', m: '_' })

const CardMeta = ({ start, duration, status }) => {
  return (
    <div className="card__meta card__meta_m-hr_top meta meta_distribute_vertical meta_m-distribute_horizontal text text_size_13-16 text_view_ghost">
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="calendar" />
          </div>
          <div className="">
            <Moment format="D MMM, HH:mm" local>
              {start}
            </Moment>
          </div>
        </div>
      </div>
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="clock" />
          </div>
          <div className="text text_size_13-16 text_view_ghost">
            {duration && prettyMilliseconds(duration, { compact: true })}
            {!duration && status}
          </div>
        </div>
      </div>
    </div>
  )
}

const CardMeta2 = ({ start, duration, status }) => {
  return (
    <div className="list__item meta meta_hr_top text text_size_13-16 text_view_ghost">
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="calendar" />
          </div>
          <div className="icon-box__content">
            <Moment format="D MMM, HH:mm" local>
              {start}
            </Moment>
          </div>
        </div>
      </div>
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="clock" />
          </div>
          <div className="text text_size_13-16 text_view_ghost">
            {duration && prettyMilliseconds(duration, { compact: true })}
            {!duration && status}
          </div>
        </div>
      </div>
    </div>
  )
}

const Card = ({ children, className, options, onClick, match }) => {
  const { id, buildNumber, commitMessage, commitHash, branchName, authorName, start, duration, status } = options
  let viewStatus = ''
  let cardMetaUnder = true
  if (status === 'Success') viewStatus = 'success'
  if (status === 'Waiting' || status === 'InProgress' || status === '') viewStatus = 'warning'
  if (status === 'Canceled' || status === 'Fail') viewStatus = 'error'

  switch (match.path) {
    case '/history':
      cardMetaUnder = false
      break
    default:
      cardMetaUnder = true
  }

  const cnCard = cn('card')
  const cnText = cn('text')
  return (
    <div onClick={onClick && ((e) => onClick(e, buildNumber, id))} className="card">
      <div className={cnCard('token')}>
        <IconBox textStyle={{ view: viewStatus }}>
          <Icon name={viewStatus} />
        </IconBox>
      </div>
      <div className={cnCard('content')}>
        <div className={cnCard('history', ['list'])}>
          <div className={cnCard('status', ['list__item'])}>
            <div
              className={`${cnCard('number')} ${cnText({ size: '18-20', view: viewStatus })}`}
            >{`#${buildNumber}`}</div>
            <div className="card__message text text_size_15-20 text_view_truncate">{commitMessage}</div>
          </div>
          <div className="list__item meta meta_m-distribute_vertical">
            <div className="meta__item">
              <div className="icon-box">
                <span className="icon icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost">
                  <Icon name="commit" />
                </span>
                <div className="icon-box__content text_size_13-16">
                  <span className="text">{branchName}</span>
                  <span className="text text_view_ghost">{commitHash.slice(0, 7)}</span>
                </div>
              </div>
            </div>
            <div className="meta__item meta__item_indent-b_8">
              <div className="icon-box">
                <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost">
                  <Icon name="user" className={{ view: 'ghost' }} />
                </div>
                <span className="text text_size_13-16">{authorName}</span>
              </div>
            </div>
          </div>
          {cardMetaUnder && CardMeta2({ duration, start, status })}
        </div>
        {!cardMetaUnder && CardMeta({ duration, start, status })}
      </div>
    </div>
  )
}

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.bool])),
}
Card.defaultProps = {
  children: '',
  className: {},
}

export default withRouter(connect()(Card))
