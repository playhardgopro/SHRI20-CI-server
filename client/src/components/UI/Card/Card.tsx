import * as React from 'react'
import { withNaming } from '@bem-react/classname'
import { useRouteMatch } from 'react-router-dom'
import Moment from 'react-moment'
import { IconBox, Icon } from '../../index'
import './Card.scss'

const prettyMilliseconds = require('pretty-ms')

const cn = withNaming({ n: '', e: '__', m: '_' })

interface CardProps {
  options: BuildTask
  onClick?(event: React.MouseEvent, { buildNumber, buildId }: { buildNumber: number; buildId: string }): void
}
interface CardMeta {
  start: string
  duration: number
  status: BuildStatus
}

const CardMeta: React.FC<CardMeta> = ({ start, duration, status }) => {
  return (
    <div className="card__meta card__meta_m-hr_top meta meta_distribute_vertical meta_m-distribute_horizontal text text_size_13-16 text_view_ghost">
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="calendar" />
          </div>
          <div className="">
            <Moment format="D MMM, HH:mm" local>
              {start}
            </Moment>
          </div>
        </div>
      </div>
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="clock" />
          </div>
          <div className="text text_size_13-16 text_view_ghost">
            {duration && prettyMilliseconds(duration, { compact: true })}
            {!duration && status}
          </div>
        </div>
      </div>
    </div>
  )
}

const CardMeta2: React.FC<CardMeta> = ({ start, duration, status }) => {
  return (
    <div className="list__item meta meta_hr_top text text_size_13-16 text_view_ghost">
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="calendar" />
          </div>
          <div className="icon-box__content">
            <Moment format="D MMM, HH:mm" local>
              {start}
            </Moment>
          </div>
        </div>
      </div>
      <div className="meta__item meta__item_indent-b_8">
        <div className="icon-box">
          <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4">
            <Icon name="clock" />
          </div>
          <div className="text text_size_13-16 text_view_ghost">
            {duration && prettyMilliseconds(duration, { compact: true })}
            {!duration && status}
          </div>
        </div>
      </div>
    </div>
  )
}

const Card: React.FC<CardProps> = ({ options, onClick }) => {
  const { id, buildNumber, commitMessage, commitHash, branchName, authorName, start, duration, status } = options
  const match = useRouteMatch()
  let viewStatus: 'success' | 'warning' | 'error' = 'error'
  let cardMetaUnder = true
  if (status === 'Success') viewStatus = 'success'
  if (status === 'Waiting' || status === 'InProgress') viewStatus = 'warning'
  if (status === 'Canceled' || status === 'Fail') viewStatus = 'error'

  switch (match.path) {
    case '/history':
      cardMetaUnder = false
      break
    default:
      cardMetaUnder = true
  }

  const cnCard = cn('card')
  const cnText = cn('text')
  return (
    <div onClick={onClick && ((e) => onClick(e, { buildNumber, buildId: id }))} className="card">
      <div className={cnCard('token')}>
        <IconBox textStyle={{ view: viewStatus }}>
          <Icon name={viewStatus} />
        </IconBox>
      </div>
      <div className={cnCard('content')}>
        <div className={cnCard('history', ['list'])}>
          <div className={cnCard('status', ['list__item'])}>
            <div
              className={`${cnCard('number')} ${cnText({ size: '18-20', view: viewStatus })}`}
            >{`#${buildNumber}`}</div>
            <div className="card__message text text_size_15-20 text_view_truncate">{commitMessage}</div>
          </div>
          <div className="list__item meta meta_m-distribute_vertical">
            <div className="meta__item">
              <div className="icon-box">
                <span className="icon icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost">
                  <Icon name="commit" />
                </span>
                <div className="icon-box__content text_size_13-16">
                  <span className="text">{branchName}</span>
                  <span className="text text_view_ghost">{commitHash.slice(0, 7)}</span>
                </div>
              </div>
            </div>
            <div className="meta__item meta__item_indent-b_8">
              <div className="icon-box">
                <div className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost">
                  <Icon name="user" className={{ view: 'ghost' }} />
                </div>
                <span className="text text_size_13-16">{authorName}</span>
              </div>
            </div>
          </div>
          {cardMetaUnder && CardMeta2({ duration, start, status })}
        </div>
        {!cardMetaUnder && CardMeta({ duration, start, status })}
      </div>
    </div>
  )
}

export default Card
