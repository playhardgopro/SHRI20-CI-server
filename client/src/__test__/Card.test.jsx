import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../components/UI/Card/Card'

jest.mock('react-router-dom', () => ({
  useRouteMatch: jest.fn(() => {
    return { path: '/history' }
  }),
}))
describe('Card', () => {
  it('history renders correctly', () => {
    const options = {
      id: '8c677446-d637-4f4c-8cca-f77baa71bd7c',
      configurationId: '262f7f07-1e2c-44f0-8931-cf4f5012c10b',
      buildNumber: 5,
      commitMessage: "Merge branch 'dz-4-react'",
      commitHash: '563db2c756a782e94c552ca86c50dfc16b558ab4',
      branchName: 'master',
      authorName: 'Daniel',
      status: 'Waiting',
      start: '2020-04-12T20:13:31.333Z',
    }
    const card = renderer.create(<Card options={options} />).toJSON()
    expect(card).toMatchSnapshot()
  })
})
