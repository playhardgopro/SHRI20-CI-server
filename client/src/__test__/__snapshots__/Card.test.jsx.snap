// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card correctly 1`] = `
<div
  className="card"
>
  <div
    className="card__token"
  >
    <div
      className="icon-box"
    >
      <div
        className="icon-box__icon text text_view_warning"
      >
        <span
          className="icon"
        >
          <svg
            height="22"
            viewBox="0 0 22 22"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 0C4.92339 0 0 4.92339 0 11C0 17.0766 4.92339 22 11 22C17.0766 22 22 17.0766 22 11C22 4.92339 17.0766 0 11 0ZM15.1024 13.8831L14.2153 14.9919C14.1571 15.0647 14.0851 15.1253 14.0034 15.1703C13.9218 15.2153 13.8321 15.2437 13.7395 15.254C13.6468 15.2643 13.5531 15.2562 13.4636 15.2303C13.3741 15.2044 13.2905 15.161 13.2177 15.1028L10.246 12.8975C10.0383 12.7313 9.87072 12.5204 9.75555 12.2807C9.64039 12.0409 9.58061 11.7783 9.58064 11.5123V4.6129C9.58064 4.42469 9.65541 4.24418 9.7885 4.11109C9.92159 3.978 10.1021 3.90323 10.2903 3.90323H11.7097C11.8979 3.90323 12.0784 3.978 12.2115 4.11109C12.3446 4.24418 12.4194 4.42469 12.4194 4.6129V11L14.9919 12.8851C15.0647 12.9433 15.1254 13.0154 15.1703 13.0971C15.2153 13.1787 15.2437 13.2685 15.254 13.3612C15.2642 13.4539 15.2561 13.5477 15.2301 13.6372C15.2041 13.7268 15.1607 13.8103 15.1024 13.8831Z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    className="card__content"
  >
    <div
      className="card__history list"
    >
      <div
        className="card__status list__item"
      >
        <div
          className="card__number text text_size_18-20 text_view_warning"
        >
          #5
        </div>
        <div
          className="card__message text text_size_15-20 text_view_truncate"
        >
          Merge branch 'dz-4-react'
        </div>
      </div>
      <div
        className="list__item meta meta_m-distribute_vertical"
      >
        <div
          className="meta__item"
        >
          <div
            className="icon-box"
          >
            <span
              className="icon icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost"
            >
              <span
                className="icon"
              >
                <svg
                  height="8"
                  viewBox="0 0 16 8"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.2 4C3.2 4.27 3.2225 4.5375 3.265 4.8H0.3C0.135 4.8 0 4.665 0 4.5V3.5C0 3.335 0.135 3.2 0.3 3.2H3.265C3.2225 3.4625 3.2 3.73 3.2 4ZM15.7 3.2H12.735C12.78 3.4625 12.8 3.73 12.8 4C12.8 4.27 12.7775 4.5375 12.735 4.8H15.7C15.865 4.8 16 4.665 16 4.5V3.5C16 3.335 15.865 3.2 15.7 3.2ZM8 1.2C7.2525 1.2 6.55 1.4925 6.02 2.02C5.49 2.55 5.2 3.2525 5.2 4C5.2 4.7475 5.49 5.45 6.02 5.98C6.55 6.5075 7.2525 6.8 8 6.8C8.7475 6.8 9.45 6.5075 9.98 5.98C10.51 5.45 10.8 4.7475 10.8 4C10.8 3.2525 10.51 2.55 9.98 2.02C9.45 1.4925 8.7475 1.2 8 1.2ZM8 0C10.21 0 12 1.79 12 4C12 6.21 10.21 8 8 8C5.79 8 4 6.21 4 4C4 1.79 5.79 0 8 0Z"
                  />
                </svg>
              </span>
            </span>
            <div
              className="icon-box__content text_size_13-16"
            >
              <span
                className="text"
              >
                master
              </span>
              <span
                className="text text_view_ghost"
              >
                563db2c
              </span>
            </div>
          </div>
        </div>
        <div
          className="meta__item meta__item_indent-b_8"
        >
          <div
            className="icon-box"
          >
            <div
              className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4 text text_view_ghost"
            >
              <span
                className="icon icon_view_ghost"
              >
                <svg
                  height="14"
                  viewBox="0 0 13 14"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.125 7C8.0582 7 9.625 5.4332 9.625 3.5C9.625 1.5668 8.0582 0 6.125 0C4.1918 0 2.625 1.5668 2.625 3.5C2.625 5.4332 4.1918 7 6.125 7ZM8.575 7.875H8.11836C7.51133 8.15391 6.83594 8.3125 6.125 8.3125C5.41406 8.3125 4.74141 8.15391 4.13164 7.875H3.675C1.64609 7.875 0 9.52109 0 11.55V12.6875C0 13.4121 0.587891 14 1.3125 14H10.9375C11.6621 14 12.25 13.4121 12.25 12.6875V11.55C12.25 9.52109 10.6039 7.875 8.575 7.875Z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="text text_size_13-16"
            >
              Daniel
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card__meta card__meta_m-hr_top meta meta_distribute_vertical meta_m-distribute_horizontal text text_size_13-16 text_view_ghost"
    >
      <div
        className="meta__item meta__item_indent-b_8"
      >
        <div
          className="icon-box"
        >
          <div
            className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4"
          >
            <span
              className="icon"
            >
              <svg
                height="16"
                viewBox="0 0 14 16"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.60806 9H3.36264C3.15714 9 2.98901 8.83125 2.98901 8.625V7.375C2.98901 7.16875 3.15714 7 3.36264 7H4.60806C4.81355 7 4.98169 7.16875 4.98169 7.375V8.625C4.98169 8.83125 4.81355 9 4.60806 9ZM7.9707 8.625V7.375C7.9707 7.16875 7.80256 7 7.59707 7H6.35165C6.14615 7 5.97802 7.16875 5.97802 7.375V8.625C5.97802 8.83125 6.14615 9 6.35165 9H7.59707C7.80256 9 7.9707 8.83125 7.9707 8.625ZM10.9597 8.625V7.375C10.9597 7.16875 10.7916 7 10.5861 7H9.34066C9.13517 7 8.96703 7.16875 8.96703 7.375V8.625C8.96703 8.83125 9.13517 9 9.34066 9H10.5861C10.7916 9 10.9597 8.83125 10.9597 8.625ZM7.9707 11.625V10.375C7.9707 10.1687 7.80256 10 7.59707 10H6.35165C6.14615 10 5.97802 10.1687 5.97802 10.375V11.625C5.97802 11.8313 6.14615 12 6.35165 12H7.59707C7.80256 12 7.9707 11.8313 7.9707 11.625ZM4.98169 11.625V10.375C4.98169 10.1687 4.81355 10 4.60806 10H3.36264C3.15714 10 2.98901 10.1687 2.98901 10.375V11.625C2.98901 11.8313 3.15714 12 3.36264 12H4.60806C4.81355 12 4.98169 11.8313 4.98169 11.625ZM10.9597 11.625V10.375C10.9597 10.1687 10.7916 10 10.5861 10H9.34066C9.13517 10 8.96703 10.1687 8.96703 10.375V11.625C8.96703 11.8313 9.13517 12 9.34066 12H10.5861C10.7916 12 10.9597 11.8313 10.9597 11.625ZM13.9487 3.5V14.5C13.9487 15.3281 13.2793 16 12.4542 16H1.49451C0.669414 16 0 15.3281 0 14.5V3.5C0 2.67188 0.669414 2 1.49451 2H2.98901V0.375C2.98901 0.16875 3.15714 0 3.36264 0H4.60806C4.81355 0 4.98169 0.16875 4.98169 0.375V2H8.96703V0.375C8.96703 0.16875 9.13517 0 9.34066 0H10.5861C10.7916 0 10.9597 0.16875 10.9597 0.375V2H12.4542C13.2793 2 13.9487 2.67188 13.9487 3.5ZM12.4542 14.3125V5H1.49451V14.3125C1.49451 14.4156 1.57857 14.5 1.68132 14.5H12.2674C12.3701 14.5 12.4542 14.4156 12.4542 14.3125Z"
                />
              </svg>
            </span>
          </div>
          <div
            className=""
          >
            <time
              dateTime={"2020-04-12T20:04:29.284Z"}
            >
              12 Apr, 23:04
            </time>
          </div>
        </div>
      </div>
      <div
        className="meta__item meta__item_indent-b_8"
      >
        <div
          className="icon-box"
        >
          <div
            className="icon-box__icon icon-box__icon_size_m icon-box__icon_indent-r_4"
          >
            <span
              className="icon"
            >
              <svg
                height="16"
                viewBox="0 0 14 16"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.2789 5.75L12.9834 5.04375C13.1299 4.89687 13.1299 4.65937 12.9834 4.5125L12.4535 3.98125C12.307 3.83438 12.07 3.83438 11.9235 3.98125L11.2783 4.62813C10.3088 3.76875 9.08371 3.19375 7.73082 3.04063V1.5H8.60365C8.80939 1.5 8.97772 1.33125 8.97772 1.125V0.375C8.97772 0.16875 8.80939 0 8.60365 0H5.3617C5.15596 0 4.98763 0.16875 4.98763 0.375V1.125C4.98763 1.33125 5.15596 1.5 5.3617 1.5H6.23454V3.04375C3.00506 3.41563 0.498779 6.1625 0.498779 9.5C0.498779 13.0906 3.40095 16 6.98268 16C10.5644 16 13.4666 13.0906 13.4666 9.5C13.4666 8.10312 13.027 6.80937 12.2789 5.75ZM6.98268 14.5C4.22702 14.5 1.99506 12.2625 1.99506 9.5C1.99506 6.7375 4.22702 4.5 6.98268 4.5C9.73833 4.5 11.9703 6.7375 11.9703 9.5C11.9703 12.2625 9.73833 14.5 6.98268 14.5ZM7.35675 11H6.60861C6.40287 11 6.23454 10.8313 6.23454 10.625V6.375C6.23454 6.16875 6.40287 6 6.60861 6H7.35675C7.56249 6 7.73082 6.16875 7.73082 6.375V10.625C7.73082 10.8313 7.56249 11 7.35675 11Z"
                />
              </svg>
            </span>
          </div>
          <div
            className="text text_size_13-16 text_view_ghost"
          >
            Waiting
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
